pkg/controller/projectclaim/projectclaim_controller.go

```diff
@@ -83,9 +80,29 @@ func (r *ReconcileProjectClaim) Reconcile(request reconcile.Request) (reconcile.
 	}
 
 	adapter := NewCustomResourceAdapter(instance, reqLogger, r.client)
+	result, err := r.ReconcileHandler(adapter)
+	if err != nil {
+		message := err.Error()
+		reason := "ReconcileFailed"
+		// Update the CRD
+		adapter.projectClaim.Status.Conditions = adapter.SetProjectClaimCondition(
```

Comment:
I don't understand why we return the condition? Doesn't the function update the projectClaim inplace?

Reply:
that was coming from my first POC. Fixed it.